arcc/arcon/src/streaming/channel/strategy/mod.rs

```diff
@@ -35,40 +29,33 @@ where
 /// Either locally through a Port or by ActorRef, or remote (ActorPath)
 fn channel_output<A, B, C>(
     channel: &Channel<A, B, C>,
-    element: ArconElement<A>,
+    event: ArconEvent<A>,
     source: *const C,
-    key: Option<u64>,
 ) -> ArconResult<()>
 where
     A: 'static + ArconType,
-    B: Port<Request = ArconElement<A>> + 'static + Clone,
+    B: Port<Request = ArconEvent<A>> + 'static + Clone,
     C: ComponentDefinition + Sized + 'static + Require<B>,
 {
     // pointer in order to escape the double borrow issue
     // TODO: find better way...
     let source = unsafe { &(*source) };
     match &channel {
         Channel::Local(actor_ref) => {
-            actor_ref.tell(Box::new(element), source);
+            actor_ref.tell(Box::new(event), source);
         }
-        Channel::Remote(actor_path) => {
-            let serialised_event: Vec<u8> = bincode::serialize(&element.data).map_err(|e| {
-                arcon_err_kind!("Failed to serialise event with err {}", e.to_string())
-            })?;
-
-            let timestamp = element.timestamp.unwrap_or(0);
-            if let Some(key) = key {
-                let keyed_msg = create_keyed_element(serialised_event, timestamp, key);
-                actor_path.tell(keyed_msg, source);
-            } else {
-                let element_msg = create_element(serialised_event, timestamp);
-                actor_path.tell(element_msg, source);
+        Channel::Remote(actor_path) => match event.to_remote() {
```

Comment:
I would suggest changing it to:
let msg = event.to_remote()?;
actor_path.tell(msg, source);

and let the caller handle the potential Error rather than printing it in the method itself.